internal/weapons/sword/filletblade/filletblade.go

```diff
@@ -0,0 +1,53 @@
+package filletblade
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+func init() {
+	core.RegisterWeaponFunc("fillet blade", weapon)
+	core.RegisterWeaponFunc("filletblade", weapon)
+}
+
+// On hit, has 50% chance to deal 240/280/320/360/400% ATK DMG to a single enemy.
+// Can only occur once every 15/14/13/12/11s.
+func weapon(char core.Character, c *core.Core, r int, param map[string]int) string {
+
+	icd := 0
+	cd := 960 - 60*r
+
+	c.Events.Subscribe(core.OnDamage, func(args ...interface{}) bool {
+		atk := args[1].(*core.AttackEvent)
+
+		if atk.Info.ActorIndex != char.CharIndex() {
+			return false
+		}
```

Comment:
missing active char check here. fillet blade does not work off-field.

Reply:
Added check